src/kernels/crashAnalysis/kernelCrashMonitor.node.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+'use strict';
+import type { KernelMessage } from '@jupyterlab/services';
+import { inject, injectable } from 'inversify';
+import { NotebookCell } from 'vscode';
+import { IExtensionSyncActivationService } from '../../platform/activation/types';
+import { IApplicationShell } from '../../platform/common/application/types';
+import { IDisposableRegistry } from '../../platform/common/types';
+import { DataScience } from '../../platform/common/utils/localize';
+import { endCellAndDisplayErrorsInCell } from '../../platform/errors/errorUtils';
+import { getDisplayNameOrNameOfKernelConnection } from '../helpers.node';
```

Comment:
This file does not have to be specific to Node, it would work perfectly fine in Web as well, in fact it should.
However `getDisplayNameOrNameOfKernelConnection` is specific to node. We'd definitely need a web version of this as well (as this is used to generate the display names for controllers).

There are number of ways to fix this, 
* Introduce an interface for this so that we can inject two different classes during runtime, one that uses web specific & other for node.
* Or get this same file to work in all cases (that might be more work)


@rchiodo which way should we go?
Ideally we can go with the latter, as all we need to get the display names are types (e.g. Python interpreter types) & reference to the `path` module. Basically it gets down to `path` module and the non-web types.
Personally I think its ok to expose the Python interpreter types into web as well, even though they are noops.

Basically types don't do anything in the webcase, i.e. we cannot have kernels with Python interpreter information, but that doesn't mean we shoudl not expose the types in web case.  I think its easier to just expose them (again a noop).

Reply:
Yeah once we have the generic vscode 'path' I think we can have this be a generic version of this file.